### ISSUE TITLE
ROOT-6.10.08-foss-2017b-Python-2.7.14.eb won't build with rpath support

### ISSUE THREAD

**Issue by casparvl** [easybuilders/easybuild-easyconfigs#7339]
During the build, CMake does a regex match between the $PATH and the compiler path. With RPATH support, the compiler path is e.g. `/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/g++_wrapper`

The + signs cause the regex match to 'not compile':

```
-- Checking whether /tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/gfortran_wrapper/gfortran supports Fortran 90
-- Checking whether /tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/gfortran_wrapper/gfortran supports Fortran 90 -- yes
RegularExpression::compile(): Nested *?+.
RegularExpression::compile(): Error in compile.
CMake Error at cmake/modules/CheckCompiler.cmake:23 (if):
  if given arguments:

    "/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/ld.bfd_wrapper:/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/ld.gold_wrapper:/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/ld_wrapper:/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/gfortran_wrapper:/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/g++_wrapper:/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/gcc_wrapper:/hpc/eb/RedHatEnterpriseServer7/pkg-config/0.29.2-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/CMake/3.9.5-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/LLVM/5.0.0-foss-2017b/bin:/hpc/eb/RedHatEnterpriseServer7/nettle/3.3-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/X11/20171023-GCCcore-6.4.0/bin:/home/casparl/.local/easybuild/RedHatEnterpriseServer7/software/fontconfig/2.12.4-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/expat/2.2.4-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/Python/2.7.14-foss-2017b/bin:/hpc/eb/RedHatEnterpriseServer7/SQLite/3.20.1-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/Tcl/8.6.7-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/libreadline/7.0-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/ncurses/6.0-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/freetype/2.8-GCCcore-6.4.0/bin:/home/casparl/.local/easybuild/RedHatEnterpriseServer7/software/libpng/1.6.32-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/cURL/7.56.1-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/PCRE/8.41-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/bzip2/1.0.6-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/libxml2/2.9.6-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/XZ/5.2.3-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/GSL/2.4-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/FFTW/3.3.6-gompi-2017b/bin:/hpc/eb/RedHatEnterpriseServer7/OpenBLAS/0.2.20-GCC-6.4.0-2.28/bin:/hpc/eb/RedHatEnterpriseServer7/OpenMPI/2.1.1-GCC-6.4.0-2.28/bin:/hpc/eb/RedHatEnterpriseServer7/hwloc/1.11.7-GCCcore-6.4.0/sbin:/hpc/eb/RedHatEnterpriseServer7/hwloc/1.11.7-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/numactl/2.0.11-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/binutils/2.28-GCCcore-6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/GCCcore/6.4.0/bin:/hpc/eb/RedHatEnterpriseServer7/EasyBuild/3.7.1/bin:/home/casparl/bin:/hpc/sw/hpc/bin:/hpc/sw/hpc/sbin:/usr/lib64/qt-3.3/bin:/hpc/sw/modules/wrappers/surfsara:/hpc/eb/modules-tcl-1.923/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/casparl/bin:/hpc/eb/utils:/hpc/eb/modules-tcl-1.923/libexec" "MATCHES" "/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/g++_wrapper"

  Regular expression
  "/tmp/casparl/eb-b6cu3o/tmpMeODrP/rpath_wrappers/g++_wrapper" cannot
  compile
Call Stack (most recent call first):
  CMakeLists.txt:43 (include)
```

Since this CheckCompiler.cmake module is ROOT-specific, this issue is really a bug in ROOT and I reported it there https://sft.its.cern.ch/jira/browse/ROOT-9883

Nevertheless, I created this issue to point others in the right direction. Also, we may wonder if it would be more robust of the automatically generated path for the g++ wrapper doesn't contain + signs, but rather would be .../rpath_wrappers/cxx_wrapper or something? I couldn't exactly figure out where this path gets its name in the framework though (and then it should really be a framework 'issue' I guess).

**Comment by boegel** [comment]
@casparvl It makes sense to fix this in framework rather than fixing it specifically for `ROOT`, since having the `+` in the path may also be problematic in other contexts.

Done in https://github.com/easybuilders/easybuild-framework/pull/2710

**Comment by boegel** [comment]
I'm closing this since we've implemented a general fix for this problem in https://github.com/easybuilders/easybuild-framework/pull/2710, to be included in EasyBuild v3.8.1
